src/runtime_src/core/tools/common/ReportMemory.cpp

```diff
@@ -225,6 +244,20 @@ populate_memtopology(const xrt_core::device * device, const std::string& desc)
   }
   pt.add_child(std::string("board.memory.data_streams"), ptStream_array);
   pt.add_child(std::string("board.memory.memories"), ptMem_array );
+   
+  // Populate softkernel mem stats 
+  boost::property_tree::ptree ptSkMem;
+  std::stringstream ss_mem(mm_buf[map->m_count + 1]);

```

Comment:
Magic number "1", why is this needed?

It might be better to define a variable the represents the size prior creating the stringstream.  Again, looking at readability.

Reply:
Fixed. Thanks.